app/serializers/api/admin/order_serializer.rb

```diff
@@ -16,6 +16,12 @@ def distributor_name
     object.distributor.andand.name
   end
 
+  def display_outstanding_balance
+    return "" unless ["balance_due", "credit_owed"].include?(object.payment_state)
+
+    "(#{object.display_outstanding_balance.to_html})"
```

Comment:
the parenthesis here are presentation, so they should be in the view not in the serializer. We may want to show this value somewhere else without the parenthesis.
So this last line becomes:
`object.display_outstanding_balance.to_html`

Reply:
@luisramos0 I'm not sure sure about this part: 

> the to_html should probably go to the view as well

`Spree::Order#display_outstanding_balance` returns a `Spree::Money` object which includes custom implementations for both `#to_s` and `#to_html`. I think in this case we probably do want to call one of these in the serializer side rather than in the view. `#to_s` is maybe the better option though. What do you think?

Agreed on the parenthesis moving to the view :+1:

Reply:
yeah, agree, one of them must be used.
I'd prefer to_s, but if it's too much trouble we can keep to_html.

Reply:
> the parenthesis here are presentation, so they should be in the view not in the serializer. We may want to show this value somewhere else without the parenthesis.
> So this last line becomes:
> `object.display_outstanding_balance`
> 

@luisramos0 Yes good point, I seem to have forgotten that this isn't a decorator for a view, it's the api response. I'll add something to the angular side of things generate the parenthesis if there is a value present.

> EDIT: the to_html should probably go to the view as well, we dont want html encoded characters in the json payload, do we?

This was me trying to see how other places use some of the spree models, I found that it's used here https://github.com/openfoodfoundation/openfoodnetwork/blob/8087e6b31b1e3a0a881cbaa0c2f67d3db45b8370/app/serializers/api/admin/order_serializer.rb#L47
But you make a good point this is silly, I should have just gone to `to_s` I was just not sure if there was some convention.

Reply:
I've made the change to use `to_s` and to add the parentheses in the angular view instead. I wasn't sure if I needed to amend my commit with these changes or make a new commit (I made a new one to not lose the comment stream)

Reply:
Sometimes the existing code is a good example, sometimes it's not :-)
I dont think it's necessarily wrong to put html encoded data in the api payload, but it seems we agree to_s is better so we can use it here and start doing that everywhere, including changing that order_serializer to_html at some point.

We dont always follow this rule but I think if a review is going on, the author should not clear the commit history so that the reviewers can see what happened. The author can clear up the commit history after the discussion is over, if it's not too much trouble.